pkg/controller.v2/jobcontroller/jobcontroller.go

```diff
@@ -26,22 +28,38 @@ import (
 // Common Interaface to be implemented by all operators
 type ControllerInterface interface {
 
-	// AdoptFunc used byControlRefManager to get the latest object if UID matches
-	AdoptFunc(job metav1.Object) func() (metav1.Object, error)
+	// Returns the Controller name
+	GetControllerName() string
```

Comment:
nit: I prefer `ControllerName()`, GetXxx() is Java style :)

Reply:
I was thinking the same. But I left it because many interfaces in kubernetes haven't followed it.
 https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/meta.go#L33 

WDYT?

Reply:
SGTM.

Reply:
@ScorpioCPH  Added the changes